### ISSUE TITLE
MethodHandle.invokeExact fails to compile on JDK 15 when targeting Java 11

### ISSUE THREAD

**Issue by srdo** [sbt/sbt#6175]
## steps

Here's a reproducer

https://github.com/srdo/MethodHandlesMiscompiled

Install Java 15.0.1 (I used Zulu).
Run `sbt clean compile` in the project root.

## problem

The Main file fails to compile.

[error] /home/stig/MethodHandlesMiscompiled/src/main/scala/Main.scala:10:27: type mismatch;
[error]  found   : Object
[error]  required: Main.TestClass
[error]         handle.invokeExact()
[error]                           ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed

## expectation

It should have compiled.

## notes
I originally opened an issue on the Scala issue tracker here https://github.com/scala/bug/issues/12236. I suspect that this may actually be an SBT or Zinc issue.

The Main file compiles fine on JDK 15 if I remove `-release 11` from the `scalacOptions`. The file compiles fine when I run `scalac -release 11 Main.scala` directly (scalac version 2.13.4). It appears to only fail to compile in SBT.

**Comment by eed3si9n** [comment]
We currently test using JDK 11, so I have no idea/expectation of how sbt or Scalac would behave under JDK 15.
I also wonder if the issue is caused by JDK 15 or by `-release 11`. sbt invokes the Scala compiler as an elaborate function call inside a sandbox ClassLoader, so I won't be surprised that some abstraction may have broken with features like `-release`.

**Comment by calm329** [comment]
@eed3si9n May I work on this issue if it's still available?

**Comment by eed3si9n** [comment]
@calm329 Yes, please. https://github.com/sbt/sbt/blob/d2cd284e5edd158d4a86ab5865b3deb787fa1542/.github/workflows/ci.yml#L147-L154

jobtype 3 runs on JDK 25, so as a starting point scripted test could be added under `java` - https://github.com/sbt/sbt/tree/develop/sbt-app/src/sbt-test/java.

**Comment by SethTisue** [comment]
for the record, I suspect that there was never a problem on the sbt side and that the bug was probably fixed by either scala/scala#9530 or scala/scala#9930
